source/common/network/listener_impl.cc

```diff
@@ -9,19 +9,53 @@
 #include "common/network/connection_impl.h"
 #include "common/ssl/connection_impl.h"
 
+#include "server/connection_handler.h"
+
 #include "event2/listener.h"
 
 namespace Network {
 
+void ListenerImpl::listenCallback(evconnlistener*, evutil_socket_t fd, sockaddr* addr, int,
+                                  void* arg) {
+  ListenerImpl* listener = static_cast<ListenerImpl*>(arg);
+
+  if (listener->use_original_dst_ && listener->connection_handler_ != nullptr) {
+    struct sockaddr_storage orig_dst_addr;
+    memset(&orig_dst_addr, 0, sizeof(orig_dst_addr));
+
+    bool success = Utility::getOriginalDst(fd, &orig_dst_addr);
+
+    if (success) {
+      std::string orig_sock_name = std::to_string(
+          listener->getAddressPort(reinterpret_cast<struct sockaddr*>(&orig_dst_addr)));
+
+      if (listener->socket_.name() != orig_sock_name) {
```

Comment:
Is this to handle the case where someone uses SO_ORIGINAL_DST but actually means to send it to the current listener? If so can you put a comment to that effect?